addons/stock_account/models/stock_location.py

```diff
@@ -9,26 +9,31 @@ class StockLocation(models.Model):
 
     valuation_in_account_id = fields.Many2one(
         'account.account', 'Stock Valuation Account (Incoming)',
-        compute='_compute_valuation_accounts', store=True, readonly=False,
+        compute='_compute_valuation_in_accounts', store=True, readonly=False,
         domain=[('account_type', 'not in', ('asset_receivable', 'liability_payable', 'asset_cash', 'liability_credit_card')), ('deprecated', '=', False)],
         help="Used for real-time inventory valuation. When set on a virtual location (non internal type), "
              "this account will be used to hold the value of products being moved from an internal location "
              "into this location, instead of the generic Stock Output Account set on the product. "
              "This has no effect for internal locations.")
     valuation_out_account_id = fields.Many2one(
         'account.account', 'Stock Valuation Account (Outgoing)',
-        compute='_compute_valuation_accounts', store=True, readonly=False,
+        compute='_compute_valuation_out_accounts', store=True, readonly=False,
         domain=[('account_type', 'not in', ('asset_receivable', 'liability_payable', 'asset_cash', 'liability_credit_card')), ('deprecated', '=', False)],
         help="Used for real-time inventory valuation. When set on a virtual location (non internal type), "
              "this account will be used to hold the value of products being moved out of this location "
              "and into an internal location, instead of the generic Stock Output Account set on the product. "
              "This has no effect for internal locations.")
 
     @api.depends('usage', 'company_id.property_stock_production_account_id')
-    def _compute_valuation_accounts(self):
+    def _compute_valuation_in_accounts(self):
         for location in self:
             if location.usage == 'production' and location.company_id:
                 location.valuation_in_account_id = location.company_id.property_stock_production_account_id
+
+    @api.depends('usage', 'company_id.property_stock_production_account_id')
+    def _compute_valuation_out_accounts(self):
+        for location in self:
+            if location.usage == 'production' and location.company_id:
```

Comment:
These computes should be in mrp_account, not in stock_account. You can extend the fields to become computed fields when mrp_account is installed.

Reply:
@ticodoo If we move this to mrp_account then the account is not set on already created locations so we need to write install post_init_hook to set the account in already created locations.
It's okay for you to write post_init_hook for this?

Reply:
@jva-odoo ah sorry I didn't see that the 'production' usage is added in stock and not mrp. You can disregard this comment